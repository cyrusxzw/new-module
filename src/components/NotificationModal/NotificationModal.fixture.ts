const NotificationModalFixture = {
  notificationMessage:
    'Sincere thanks for subscribing. We are always pleased to share ourpassionate interests with like - minded individuals.',
};

export { NotificationModalFixture };
