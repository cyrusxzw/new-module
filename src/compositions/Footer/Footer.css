@import 'styles/index';

.base {
  position: relative;
  background-color: var(--color-grey-10);
}

.block {
  display: grid;
  display: -ms-grid;
  padding: 35px 20px;
  border-bottom: 3px solid #d6d5cb;
  grid-column-gap: 20px;
  grid-row-gap: 50px;
  grid-template-columns: repeat(2, 1fr);
  grid-template-rows: auto 1fr;
  @media (--viewport-md) {
    padding: 40px 25px;
    grid-column-gap: 35px;
    grid-row-gap: 35px;
    grid-template-columns: repeat(3, 1fr);
    grid-template-rows: auto 1fr;
  }
  @media (--viewport-lg) {
    padding: 50px 40px;
    grid-column-gap: 40px;
    grid-row-gap: 15px;
    grid-template-columns: repeat(5, 1fr);
    grid-template-rows: auto 1fr;
  }
}

.bottom {
  display: flex;
  height: 72px;
  align-items: center;
  justify-content: space-between;
  padding-right: 20px;
  padding-left: 20px;
  color: var(--color-light-copy);
  font-family: var(--font-family-sans-serif-regular);
  @media (--viewport-md) {
    padding-right: 25px;
    padding-left: 25px;
  }
  @media (--viewport-lg) {
    padding-right: 40px;
    padding-left: 40px;
  }
}

.signup {
  grid-column: 1 / 3;
  @media (--viewport-md-only) {
    grid-column: 1 / 4;
  }
  @media (--viewport-lg) {
    margin-top: 10px;
  }
}

.item {
  grid-column: 1 span;
  @media (--viewport-lg) {
    &:nth-of-type(4n + 0) {
      grid-column: 2 span;
    }
  }
}

.iconLinks {
  margin-left: 45px;
}

/* Handle the grid css to support IE */
@media all and (-ms-high-contrast: none) {
  @media (--viewport-xs-sm-only) {
    .block {
      -ms-grid-columns: 1fr 20px 1fr;
      -ms-grid-rows: 1fr 50px 1fr 50px 1fr 50px 1fr;
    }

    .signup {
      -ms-grid-column: 1;
      -ms-grid-column-span: 3;
    }

    .item:nth-of-type(2n + 0) {
      -ms-grid-column: 3;
      -ms-grid-column-span: 1;
    }

    .item:nth-of-type(2n + 1) {
      -ms-grid-column: 1;
      -ms-grid-column-span: 1;
    }

    .item:nth-of-type(-n + 2) {
      -ms-grid-row: 3;
      -ms-grid-row-span: 1;
    }

    .item:nth-of-type(3),
    .item:nth-of-type(4) {
      -ms-grid-row: 5;
      -ms-grid-row-span: 1;
    }

    .item:nth-of-type(5),
    .item:nth-of-type(6) {
      -ms-grid-row: 7;
      -ms-grid-row-span: 1;
    }
  }

  @media (--viewport-md-only) {
    .block {
      -ms-grid-columns: 1fr 35px 1fr 35px 1fr;
      -ms-grid-rows: 1fr 35px 1fr 35px 1fr;
    }

    .signup {
      -ms-grid-column: 1;
      -ms-grid-column-span: 5;
    }

    .item:nth-of-type(3n + 0) {
      -ms-grid-column: 5;
      -ms-grid-column-span: 1;
    }

    .item:nth-of-type(3n + 1) {
      -ms-grid-column: 1;
      -ms-grid-column-span: 1;
    }

    .item:nth-of-type(3n + 2) {
      -ms-grid-column: 3;
      -ms-grid-column-span: 1;
    }

    .item:nth-of-type(-n + 3) {
      -ms-grid-row: 3;
      -ms-grid-row-span: 1;
    }

    .item:nth-of-type(n + 4) {
      -ms-grid-row: 5;
      -ms-grid-row-span: 1;
    }
  }

  @media (--viewport-lg) {
    .block {
      -ms-grid-columns: 2fr 40px 1fr 40px 1fr 40px 1fr;
      -ms-grid-rows: 1fr 15px 1fr;
    }

    .signup {
      margin-top: 10px;
      -ms-grid-column: 1;
      -ms-grid-column-span: 1;
    }

    .item:nth-of-type(4n + 1) {
      -ms-grid-column: 3;
      -ms-grid-column-span: 1;
    }

    .item:nth-of-type(4n + 2) {
      -ms-grid-column: 5;
      -ms-grid-column-span: 1;
    }

    .item:nth-of-type(4n + 3) {
      -ms-grid-column: 7;
      -ms-grid-column-span: 1;
    }

    .item:nth-of-type(n + 4) {
      -ms-grid-row: 3;
      -ms-grid-row-span: 1;
    }
  }
}
