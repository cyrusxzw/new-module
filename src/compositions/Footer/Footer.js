import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import { NotificationContextProvider } from '~/contexts';
import { NewsletterSignUp } from '~/compositions/NewsletterSignUp/index.ts';
import { FooterBlock } from '~/components/FooterBlock/index.ts';
import { Hidden } from '~/components/Hidden/index.ts';
import { IconLink } from '~/components/IconLink/index.ts';
import { NotificationModal } from '~/components/NotificationModal/index.ts';
import styles from './Footer.css';

const Footer = ({
  blocks,
  className,
  consentErrorMsg,
  iconLinks,
  errorMessage,
  notificationMessage,
  showTermsConditionsTextBox,
  subscriptionMessage,
  termsAndCondition,
  termsMessage,
  onClick,
}) => {
  const baseClassSet = cx(styles.base, className);

  return (
    <NotificationContextProvider>
      <div className={baseClassSet}>
        <NotificationModal
          backgroundColor="#d5d5cc"
          notificationMessage={notificationMessage}
        />
        <div className={styles.block}>
          <div className={styles.signup}>
            <NewsletterSignUp
              consentErrorMsg={consentErrorMsg}
              errorMessage={errorMessage}
              onClick={onClick}
              showTermsConditionsTextBox={showTermsConditionsTextBox}
              subscriptionMessage={subscriptionMessage}
              termsAndCondition={termsAndCondition}
              termsMessage={termsMessage}
              theme="light"
            />
          </div>
          {blocks?.length &&
            blocks.map((block, index) => (
              <FooterBlock
                className={styles.item}
                copy={block.copy}
                heading={block.heading}
                isVisibleOnTabletAndMobile={block.isVisibleOnTabletAndMobile}
                key={index}
                links={block.links}
                theme="light"
              />
            ))}
        </div>
        <div className={styles.bottom}>
          <span>&copy; Aesop</span>
          <Hidden isLarge={true} isXLarge={true}>
            <div>
              {iconLinks?.length &&
                iconLinks.map(icon => (
                  <IconLink
                    altText={icon.altText}
                    className={styles.iconLinks}
                    hasTargetInNewWindow={icon.hasTargetInNewWindow}
                    height={21}
                    icon={icon.icon}
                    id={icon.altText}
                    key={icon.id}
                    target={icon.target}
                    width={21}
                  />
                ))}
            </div>
          </Hidden>
        </div>
      </div>
    </NotificationContextProvider>
  );
};

Footer.propTypes = {
  blocks: PropTypes.arrayOf(
    PropTypes.shape({
      copy: PropTypes.string,
      heading: PropTypes.string,
      isVisibleOnTabletAndMobile: PropTypes.bool,
      links: PropTypes.array,
    }),
  ),
  className: PropTypes.string,
  consentErrorMsg: PropTypes.string,
  iconLinks: PropTypes.array,
  errorMessage: PropTypes.string,
  notificationMessage: PropTypes.string,
  showTermsConditionsTextBox: PropTypes.bool,
  subscriptionMessage: PropTypes.string,
  termsAndCondition: PropTypes.string,
  termsMessage: PropTypes.string,
  onClick: PropTypes.func,
};

Footer.defaultProps = {
  blocks: undefined,
  className: undefined,
  consentErrorMsg: undefined,
  iconLinks: undefined,
  errorMessage: undefined,
  notificationMessage: undefined,
  showTermsConditionsTextBox: undefined,
  subscriptionMessage: undefined,
  termsAndCondition: undefined,
  termsMessage: undefined,
  onClick: undefined,
};

export { Footer };
