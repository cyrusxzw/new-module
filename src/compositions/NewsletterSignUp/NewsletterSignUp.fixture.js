const NewsletterSignUpFixture = {
  consentErrorMsg:
    'Please confirm your accept our Terms & Conditions before proceeding ',
  errorMessage: 'Please enter valid email address',
  termsAndCondition:
    'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Quisque quis sem quis sem bibendum aliquet non nec sapien. Suspendisse felis dui, gravida non ligula id, tempor pharetra sapien. Vestibulum porttitor felis vitae mollis porta. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Quisque quis sem quis sem bibendum aliquet non nec sapien. Suspendisse felis dui, gravida non ligula id, tempor pharetra sapien. Vestibulum porttitor felis vitae mollis porta. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Quisque quis sem quis sem bibendum aliquet non nec sapien. Suspendisse felis dui, gravida non ligula id, tempor pharetra sapien. Vestibulum porttitor felis vitae mollis porta. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Quisque quis sem quis sem bibendum aliquet non nec sapien. Suspendisse felis dui, gravida non ligula id, tempor pharetra sapien. Vestibulum porttitor felis vitae mollis porta.',
  termsMessage: 'I accept the Terms & Conditions ',
  subscriptionMessage:
    'By subscribing, you agree to receive email marketing from Aesop about our products and services, and accept our',
};

export { NewsletterSignUpFixture };
