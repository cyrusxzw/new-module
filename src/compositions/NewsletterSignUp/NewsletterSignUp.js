import React, { useState } from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import { useNotificationContext } from '~/contexts';
import { Icon } from '~/components/Icon';
import { Loading } from '~/components/Loading';
import { Checkbox } from '~/components/Checkbox';
import { TextInput } from '~/components/TextInput/index.ts';
import styles from './NewsletterSignUp.css';

const NewsletterSignUp = ({
  className,
  consentErrorMsg,
  errorMessage,
  onClick,
  showTermsConditionsTextBox,
  termsAndCondition,
  termsMessage,
  theme,
  subscriptionMessage,
}) => {
  const [hasContent, setHasContent] = useState(false);
  const [email, setEmail] = useState(undefined);
  const [isLoading, setIsLoading] = useState(false);
  const [error, setError] = useState(undefined);
  const [hasAcceptedSubscription, setHasAcceptedSubscription] = useState(false);
  const [hasAcceptedTerms, setHasAcceptedTerms] = useState(false);

  const inputFieldClassSet = cx(styles.inputField, styles[theme], className);
  const buttonClassSet = cx(
    styles.button,
    styles[theme],
    !!error && styles.hasError,
  );
  const signUpIconClassSet = cx(styles.signUpIcon, styles[theme]);
  const loadingClassSet = cx(styles.loading, styles[theme]);
  const subscriptionMessageClassSet = cx(
    styles.subscriptionMessage,
    styles[theme],
  );

  const footerSuccessModal = useNotificationContext();
  const { actionType, dispatch } = footerSuccessModal;

  const validate = (value) => {
    const EMAIL_REGEXP = /^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,}$/i;
    const string = String(value);
    if (!string) {
      return true;
    }

    const isValid = !!string.match(EMAIL_REGEXP);
    return isValid;
  };

  const onChange = (e) => {
    setEmail(e.target.value);
    setHasContent(e.target.value && e.target.value.trim() !== '');
  };

  const handleSubmit = async (e) => {
    e.preventDefault();
    setError(undefined);
    const isValid = validate(email);
    if (
      isValid &&
      !(
        showTermsConditionsTextBox &&
        (!hasAcceptedSubscription || !hasAcceptedTerms)
      )
    ) {
      setIsLoading(true);
      await onClick(email);
      setIsLoading(false);
      dispatch({ type: actionType });
    } else {
      let errorMsg = errorMessage;
      if (
        showTermsConditionsTextBox &&
        (!hasAcceptedSubscription || !hasAcceptedTerms)
      ) {
        errorMsg = consentErrorMsg;
      }
      setError(errorMsg);
    }
  };

  return (
    <div className={styles.base}>
      <form onSubmit={handleSubmit}>
        <div className={inputFieldClassSet}>
          <div className={styles.footerNewsletterInputWrapper}>
            <TextInput
              data-test-ref="EMAIL_SIGN_UP_INPUT"
              hasContent={hasContent}
              hasError={!!error}
              id="newsletter-email"
              inputClassName={cx(styles.footerNewsletter)}
              isEnabled={!isLoading}
              label="Email Address"
              name="newsletter-email"
              onChange={onChange}
              theme={theme}
            />
          </div>
          <button
            className={buttonClassSet}
            data-test-ref="EMAIL_SIGN_UP_BUTTON"
            disabled={!!isLoading || !hasContent}
          >
            <Icon
              className={signUpIconClassSet}
              name="rightArrow"
              theme={theme}
            />
          </button>
          {!!isLoading && (
            <Loading
              className={loadingClassSet}
              isLoading={true}
              theme={theme}
            />
          )}
        </div>
        <div className={styles.footerNewsletterErrorMessage}>{error}</div>
        {showTermsConditionsTextBox ? (
          <section>
            <Checkbox
              content={subscriptionMessage}
              dataTestRef="EMAIL_SIGN_UP_SUBSCRIPTION_CHECKBOX"
              id="subscription"
              isEnabled={!isLoading}
              onChange={() =>
                setHasAcceptedSubscription(!hasAcceptedSubscription)
              }
              theme={theme}
            />
            <Checkbox
              content={termsMessage}
              dataTestRef="EMAIL_SIGN_UP_TERMS_AND_CONDITIONS_CHECKBOX"
              id="terms"
              isEnabled={!isLoading}
              onChange={() => setHasAcceptedTerms(!hasAcceptedTerms)}
              theme={theme}
            />
            <div className={styles.termsAndConditionWrapper}>
              <div className={styles.termsAndCondition}>
                {termsAndCondition}
              </div>
            </div>
          </section>
        ) : (
          <div className={subscriptionMessageClassSet}>
            {subscriptionMessage}
          </div>
        )}
      </form>
    </div>
  );
};

NewsletterSignUp.propTypes = {
  className: PropTypes.string,
  consentErrorMsg: PropTypes.string,
  errorMessage: PropTypes.string,
  onClick: PropTypes.func,
  showTermsConditionsTextBox: PropTypes.bool,
  termsAndCondition: PropTypes.string,
  termsMessage: PropTypes.string,
  theme: PropTypes.string,
  subscriptionMessage: PropTypes.string,
};

NewsletterSignUp.defaultProps = {
  className: undefined,
  consentErrorMsg: undefined,
  errorMessage: 'NEWSLETTER_SIGNUP',
  onClick: undefined,
  showTermsConditionsTextBox: false,
  termsAndCondition: undefined,
  termsMessage: undefined,
  theme: 'dark',
  subscriptionMessage: undefined,
};

export { NewsletterSignUp };
