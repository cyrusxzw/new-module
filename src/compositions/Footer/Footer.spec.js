import React from 'react';
import { render } from '@testing-library/react';
import { Footer } from './Footer';
import { FooterFixture } from './Footer.fixture';
import { NewsletterSignUpFixture } from '../NewsletterSignUp/NewsletterSignUp.fixture';
import { NotificationModalFixture } from '~/components/NotificationModal/NotificationModal.fixture';

describe('<Footer />', () => {
  it('should render base component correctly', () => {
    const { container } = render(
      <Footer
        blocks={FooterFixture.blocks}
        consentErrorMsg={NewsletterSignUpFixture.consentErrorMsg}
        errorMessage={NewsletterSignUpFixture.errorMessage}
        notificationMessage={NotificationModalFixture.notificationMessage}
        showTermsConditionsTextBox={true}
        subscriptionMessage={NewsletterSignUpFixture.subscriptionMessage}
        termsAndCondition={NewsletterSignUpFixture.termsAndCondition}
        termsMessage={NewsletterSignUpFixture.termsMessage}
      />,
    );

    expect(container).toMatchSnapshot();
  });
});
