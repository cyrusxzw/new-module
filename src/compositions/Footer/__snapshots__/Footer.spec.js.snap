// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should render base component correctly 1`] = `
<div>
  <div
    class="base"
  >
    <div
      class="base isNotVisible"
      style="background-color: rgb(213, 213, 204);"
    >
      <span
        class="notificationMessage"
      >
        Sincere thanks for subscribing. We are always pleased to share ourpassionate interests with like - minded individuals.
      </span>
      <button
        class="closeButton"
        data-ref="close"
        tabindex="-1"
      >
        <svg
          aria-labelledby="close-1"
          class="base close dark"
          data-testid="data-testid-Icon"
          focusable="false"
          height="15"
          role="img"
          style="height: 15px; width: 15px;"
          tabindex="-1"
          viewBox="0 0 50 50"
          width="15"
        >
          <title
            id="close-1"
          >
            close
          </title>
          <g>
            <polygon
              class="close"
              points="50,5 45,0 25,20 5,0 0,5 20,25 0,45 5,50 25,30 45,50 50,45 30,25"
            />
          </g>
        </svg>
      </button>
    </div>
    <div
      class="block"
    >
      <div
        class="signup"
      >
        <div
          class="base"
        >
          <form>
            <div
              class="inputField light"
            >
              <div
                class="footerNewsletterInputWrapper"
              >
                <div
                  class="formText"
                >
                  <label
                    for="newsletter-email"
                  >
                    <input
                      class="formTextInput light footerNewsletter"
                      data-test-ref="EMAIL_SIGN_UP_INPUT"
                      id="newsletter-email"
                      name="newsletter-email"
                      novalidate=""
                      type="text"
                      value=""
                    />
                    <span
                      class="formTextLabel light"
                    >
                      Email Address
                    </span>
                  </label>
                </div>
              </div>
              <button
                class="button light"
                data-test-ref="EMAIL_SIGN_UP_BUTTON"
                disabled=""
              >
                <svg
                  aria-labelledby="rightArrow-2"
                  class="base rightArrow light signUpIcon light"
                  data-testid="data-testid-Icon"
                  focusable="false"
                  height="12"
                  role="img"
                  style="height: 12px; width: 12px;"
                  tabindex="-1"
                  viewBox="0 0 50 50"
                  width="12"
                >
                  <title
                    id="rightArrow-2"
                  >
                    rightArrow
                  </title>
                  <g>
                    <path
                      d="M30.1,5.3L50,25.1L30.1,45h-6.6l18-17.6H0v-4.8h41.5l-18-17.6h6.6V5.3z"
                    />
                  </g>
                </svg>
              </button>
            </div>
            <div
              class="footerNewsletterErrorMessage"
            />
            <section>
              <div>
                <label
                  class="base"
                  for="subscription"
                >
                  <input
                    aria-describedby="subscription-error-message"
                    aria-invalid="false"
                    class="checkBox light"
                    data-test-ref="EMAIL_SIGN_UP_SUBSCRIPTION_CHECKBOX"
                    id="subscription"
                    type="checkbox"
                  />
                  <span
                    class="content light"
                  >
                    By subscribing, you agree to receive email marketing from Aesop about our products and services, and accept our
                  </span>
                </label>
              </div>
              <div>
                <label
                  class="base"
                  for="terms"
                >
                  <input
                    aria-describedby="terms-error-message"
                    aria-invalid="false"
                    class="checkBox light"
                    data-test-ref="EMAIL_SIGN_UP_TERMS_AND_CONDITIONS_CHECKBOX"
                    id="terms"
                    type="checkbox"
                  />
                  <span
                    class="content light"
                  >
                    I accept the Terms & Conditions 
                  </span>
                </label>
              </div>
              <div
                class="termsAndConditionWrapper"
              >
                <div
                  class="termsAndCondition"
                >
                  Lorem ipsum dolor sit amet, consectetur adipiscing elit. Quisque quis sem quis sem bibendum aliquet non nec sapien. Suspendisse felis dui, gravida non ligula id, tempor pharetra sapien. Vestibulum porttitor felis vitae mollis porta. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Quisque quis sem quis sem bibendum aliquet non nec sapien. Suspendisse felis dui, gravida non ligula id, tempor pharetra sapien. Vestibulum porttitor felis vitae mollis porta. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Quisque quis sem quis sem bibendum aliquet non nec sapien. Suspendisse felis dui, gravida non ligula id, tempor pharetra sapien. Vestibulum porttitor felis vitae mollis porta. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Quisque quis sem quis sem bibendum aliquet non nec sapien. Suspendisse felis dui, gravida non ligula id, tempor pharetra sapien. Vestibulum porttitor felis vitae mollis porta.
                </div>
              </div>
            </section>
          </form>
        </div>
      </div>
      <section
        class="base light item"
      >
        <h3
          class="base mediumWeightFont xXSmall light heading"
        >
          Orders & Support
        </h3>
        <div
          class="divider light"
        />
        <ul
          class="list"
        >
          <li>
            <a
              class="base hasIcon hasExternalIcon left light listItem"
              data-test-ref="FOOTER_LINK"
              href="https://www.aesop.com/au/r/contact-us"
              target="_blank"
              title="Contact us"
            >
              Contact us
               
              <i
                aria-hidden="true"
                class="icon"
              >
                <svg
                  aria-labelledby="rightUpArrow-5"
                  class="base rightUpArrow light"
                  data-testid="data-testid-Icon"
                  focusable="false"
                  height="12"
                  role="img"
                  style="height: 12px; width: 12px;"
                  tabindex="-1"
                  viewBox="0 0 50 50"
                  width="12"
                >
                  <title
                    id="rightUpArrow-5"
                  >
                    rightUpArrow
                  </title>
                  <g>
                    <path
                      d="M21.8,0.2l28.1-0.1v28.1l-4.5,0l0.1-20.5L16.2,37.1l-3.4-3.4L42.1,4.4L21.8,4.4L21.8,0.2L21.8,0.2z"
                    />
                  </g>
                </svg>
              </i>
            </a>
          </li>
          <li>
            <a
              class="base hasIcon hasExternalIcon left light listItem"
              data-test-ref="FOOTER_LINK"
              href="https://www.aesop.com/au/r/shipping/"
              target="_blank"
              title="Delivery and returns"
            >
              Delivery and returns
               
              <i
                aria-hidden="true"
                class="icon"
              >
                <svg
                  aria-labelledby="rightUpArrow-6"
                  class="base rightUpArrow light"
                  data-testid="data-testid-Icon"
                  focusable="false"
                  height="12"
                  role="img"
                  style="height: 12px; width: 12px;"
                  tabindex="-1"
                  viewBox="0 0 50 50"
                  width="12"
                >
                  <title
                    id="rightUpArrow-6"
                  >
                    rightUpArrow
                  </title>
                  <g>
                    <path
                      d="M21.8,0.2l28.1-0.1v28.1l-4.5,0l0.1-20.5L16.2,37.1l-3.4-3.4L42.1,4.4L21.8,4.4L21.8,0.2L21.8,0.2z"
                    />
                  </g>
                </svg>
              </i>
            </a>
          </li>
          <li>
            <a
              class="base hasIcon hasExternalIcon left light listItem"
              data-test-ref="FOOTER_LINK"
              href="https://www.aesop.com/au/r/contact-us"
              target="_blank"
              title="Click And Collect"
            >
              Click And Collect
               
              <i
                aria-hidden="true"
                class="icon"
              >
                <svg
                  aria-labelledby="rightUpArrow-7"
                  class="base rightUpArrow light"
                  data-testid="data-testid-Icon"
                  focusable="false"
                  height="12"
                  role="img"
                  style="height: 12px; width: 12px;"
                  tabindex="-1"
                  viewBox="0 0 50 50"
                  width="12"
                >
                  <title
                    id="rightUpArrow-7"
                  >
                    rightUpArrow
                  </title>
                  <g>
                    <path
                      d="M21.8,0.2l28.1-0.1v28.1l-4.5,0l0.1-20.5L16.2,37.1l-3.4-3.4L42.1,4.4L21.8,4.4L21.8,0.2L21.8,0.2z"
                    />
                  </g>
                </svg>
              </i>
            </a>
          </li>
          <li>
            <a
              class="base left light listItem"
              data-test-ref="FOOTER_LINK"
              href="https://www.aesop.com/au/r/shipping/"
              target="_self"
              title="Track Your Orders"
            >
              Track Your Orders
            </a>
          </li>
        </ul>
      </section>
      <section
        class="base light item"
      >
        <h3
          class="base mediumWeightFont xXSmall light heading"
        >
          Services
        </h3>
        <div
          class="divider light"
        />
        <ul
          class="list"
        >
          <li>
            <a
              class="base left light listItem"
              data-test-ref="FOOTER_LINK"
              href="https://www.aesop.com/au/r/contact-us"
              target="_self"
              title="Video Consultation"
            >
              Video Consultation
            </a>
          </li>
          <li>
            <a
              class="base left light listItem"
              data-test-ref="FOOTER_LINK"
              href="https://www.aesop.com/au/r/shipping/"
              target="_self"
              title="Live Assistance"
            >
              Live Assistance
            </a>
          </li>
          <li>
            <a
              class="base left light listItem"
              data-test-ref="FOOTER_LINK"
              href="https://www.aesop.com/au/r/contact-us"
              target="_self"
              title="Call And Collect"
            >
              Call And Collect
            </a>
          </li>
          <li>
            <a
              class="base left light listItem"
              data-test-ref="FOOTER_LINK"
              href="https://www.aesop.com/au/r/shipping/"
              target="_self"
              title="Facial Appointment"
            >
              Facial Appointment
            </a>
          </li>
          <li>
            <a
              class="base left light listItem"
              data-test-ref="FOOTER_LINK"
              href="https://www.aesop.com/au/r/shipping/"
              target="_self"
              title="Corporate Gifts"
            >
              Corporate Gifts
            </a>
          </li>
        </ul>
      </section>
      <section
        class="base light item"
      >
        <h3
          class="base mediumWeightFont xXSmall light heading"
        >
          Shipping to
        </h3>
        <div
          class="divider light"
        />
        <div
          class="listItem copy"
        >
          TODO: Shipping To Component
        </div>
      </section>
      <section
        class="base light item"
      >
        <h3
          class="base mediumWeightFont xXSmall light heading"
        >
          Acknowledgment of Country
        </h3>
        <div
          class="divider light"
        />
        <div
          class="listItem copy"
        >
          Aesop acknowledges the Traditional Owners of the lands on which we live and work, and pays deepest respects to Elders past, present and emerging. We also acknowledge that sovereignty of these lands has never been ceded.
        </div>
      </section>
      <section
        class="base light item"
      >
        <h3
          class="base mediumWeightFont xXSmall light heading"
        >
          Sustainability
        </h3>
        <div
          class="divider light"
        />
        <div
          class="listItem copy"
        >
          Our approach to sustainability is guided by where we can have the greatest social and environmental impact.
        </div>
      </section>
      <section
        class="base light item"
      >
        <h3
          class="base mediumWeightFont xXSmall light heading"
        >
          About
        </h3>
        <div
          class="divider light"
        />
        <ul
          class="list"
        >
          <li>
            <a
              class="base left light listItem"
              data-test-ref="FOOTER_LINK"
              href="https://www.aesop.com/au/r/contact-us"
              target="_self"
              title="About Us"
            >
              About Us
            </a>
          </li>
          <li>
            <a
              class="base left light listItem"
              data-test-ref="FOOTER_LINK"
              href="https://www.aesop.com/au/r/shipping/"
              target="_self"
              title="Career"
            >
              Career
            </a>
          </li>
          <li>
            <a
              class="base left light listItem"
              data-test-ref="FOOTER_LINK"
              href="https://www.aesop.com/au/r/contact-us"
              target="_self"
              title="Press"
            >
              Press
            </a>
          </li>
          <li>
            <a
              class="base left light listItem"
              data-test-ref="FOOTER_LINK"
              href="https://www.aesop.com/au/r/shipping/"
              target="_self"
              title="Policy"
            >
              Policy
            </a>
          </li>
          <li>
            <a
              class="base left light listItem"
              data-test-ref="FOOTER_LINK"
              href="https://www.aesop.com/au/r/shipping/"
              target="_self"
              title="Privacy Policy"
            >
              Privacy Policy
            </a>
          </li>
          <li>
            <a
              class="base left light listItem"
              data-test-ref="FOOTER_LINK"
              href="https://www.aesop.com/au/r/shipping/"
              target="_self"
              title="Accessibility"
            >
              Accessibility
            </a>
          </li>
          <li>
            <a
              class="base left light listItem"
              data-test-ref="FOOTER_LINK"
              href="https://www.aesop.com/au/r/shipping/"
              target="_self"
              title="Terms and Conditions"
            >
              Terms and Conditions
            </a>
          </li>
        </ul>
      </section>
      <section
        class="base light item"
      >
        <h3
          class="base mediumWeightFont xXSmall light heading"
        >
          Social Media
        </h3>
        <div
          class="divider light"
        />
        <ul
          class="list"
        >
          <li>
            <a
              class="base hasIcon hasExternalIcon left light listItem"
              data-test-ref="FOOTER_LINK"
              href="https://www.aesop.com/au/r/contact-us"
              target="_blank"
              title="Instagram"
            >
              Instagram
               
              <i
                aria-hidden="true"
                class="icon"
              >
                <svg
                  aria-labelledby="rightUpArrow-8"
                  class="base rightUpArrow light"
                  data-testid="data-testid-Icon"
                  focusable="false"
                  height="12"
                  role="img"
                  style="height: 12px; width: 12px;"
                  tabindex="-1"
                  viewBox="0 0 50 50"
                  width="12"
                >
                  <title
                    id="rightUpArrow-8"
                  >
                    rightUpArrow
                  </title>
                  <g>
                    <path
                      d="M21.8,0.2l28.1-0.1v28.1l-4.5,0l0.1-20.5L16.2,37.1l-3.4-3.4L42.1,4.4L21.8,4.4L21.8,0.2L21.8,0.2z"
                    />
                  </g>
                </svg>
              </i>
            </a>
          </li>
          <li>
            <a
              class="base hasIcon hasExternalIcon left light listItem"
              data-test-ref="FOOTER_LINK"
              href="https://www.aesop.com/au/r/contact-us"
              target="_blank"
              title="Twitter"
            >
              Twitter
               
              <i
                aria-hidden="true"
                class="icon"
              >
                <svg
                  aria-labelledby="rightUpArrow-9"
                  class="base rightUpArrow light"
                  data-testid="data-testid-Icon"
                  focusable="false"
                  height="12"
                  role="img"
                  style="height: 12px; width: 12px;"
                  tabindex="-1"
                  viewBox="0 0 50 50"
                  width="12"
                >
                  <title
                    id="rightUpArrow-9"
                  >
                    rightUpArrow
                  </title>
                  <g>
                    <path
                      d="M21.8,0.2l28.1-0.1v28.1l-4.5,0l0.1-20.5L16.2,37.1l-3.4-3.4L42.1,4.4L21.8,4.4L21.8,0.2L21.8,0.2z"
                    />
                  </g>
                </svg>
              </i>
            </a>
          </li>
          <li>
            <a
              class="base hasIcon hasExternalIcon left light listItem"
              data-test-ref="FOOTER_LINK"
              href="https://www.aesop.com/au/r/contact-us"
              target="_blank"
              title="KaKao Plus Friends"
            >
              KaKao Plus Friends
               
              <i
                aria-hidden="true"
                class="icon"
              >
                <svg
                  aria-labelledby="rightUpArrow-10"
                  class="base rightUpArrow light"
                  data-testid="data-testid-Icon"
                  focusable="false"
                  height="12"
                  role="img"
                  style="height: 12px; width: 12px;"
                  tabindex="-1"
                  viewBox="0 0 50 50"
                  width="12"
                >
                  <title
                    id="rightUpArrow-10"
                  >
                    rightUpArrow
                  </title>
                  <g>
                    <path
                      d="M21.8,0.2l28.1-0.1v28.1l-4.5,0l0.1-20.5L16.2,37.1l-3.4-3.4L42.1,4.4L21.8,4.4L21.8,0.2L21.8,0.2z"
                    />
                  </g>
                </svg>
              </i>
            </a>
          </li>
          <li>
            <a
              class="base hasIcon hasExternalIcon left light listItem"
              data-test-ref="FOOTER_LINK"
              href="https://www.aesop.com/au/r/contact-us"
              target="_blank"
              title="Naver Post"
            >
              Naver Post
               
              <i
                aria-hidden="true"
                class="icon"
              >
                <svg
                  aria-labelledby="rightUpArrow-11"
                  class="base rightUpArrow light"
                  data-testid="data-testid-Icon"
                  focusable="false"
                  height="12"
                  role="img"
                  style="height: 12px; width: 12px;"
                  tabindex="-1"
                  viewBox="0 0 50 50"
                  width="12"
                >
                  <title
                    id="rightUpArrow-11"
                  >
                    rightUpArrow
                  </title>
                  <g>
                    <path
                      d="M21.8,0.2l28.1-0.1v28.1l-4.5,0l0.1-20.5L16.2,37.1l-3.4-3.4L42.1,4.4L21.8,4.4L21.8,0.2L21.8,0.2z"
                    />
                  </g>
                </svg>
              </i>
            </a>
          </li>
        </ul>
      </section>
    </div>
    <div
      class="bottom"
    >
      <span>
        © Aesop
      </span>
      <div />
    </div>
  </div>
</div>
`;
